### ISSUE TITLE
[DevTools Bug]: Components + Profile tabs not showing up in Chrome 99

### ISSUE THREAD

**Issue by scott-m-sarsfield** [facebook/react#24094]
### Website or app

https://codesandbox.io/s/new

### Repro steps

1. Open up Chrome inspector
2. Expect Components + Profile tabs to appear

Computer:
2019 MacBook Pro (Intel)

Browser:
Google Chrome
Version 99.0.4844.51 (Official Build) (x86_64)

DevTools:
4.24.0 (3/10/2022)

### How often does this bug happen?

Every time

**Comment by scott-m-sarsfield** [comment]
Seems like I'm not the only one: https://www.reddit.com/r/reactjs/comments/ta820t/react_devtools_extension_not_working_on_my/

**Comment by icruzr93** [comment]
I'm facing exactly the same problem but with all my extensions not only with this one.

**Comment by lunaruan** [comment]
Hey! Thanks for reporting. This sounds less like a DevTools issue and more like a Chrome issue, especially since all extensions seem to be having this issue. You can report issues with Chrome [here](https://support.google.com/chrome/answer/95315?hl=en&co=GENIE.Platform%3DDesktop). Closing this issue!

**Comment by scott-m-sarsfield** [comment]
saga continues:
https://bugs.chromium.org/p/chromium/issues/detail?id=1306249

**Comment by scott-m-sarsfield** [comment]
Ok.  Somehow I got them working again (although I don't have the most reliable procedure for reproducing).

I turned off two flags in the devtool experiments section:  and `Enable Reporting API panel in the Application panel`.  I also thought that doing a refresh of devtools could have worked, but it didn't appear to be consistent.  I think turning off the first flag is probably the only thing, but I'm not 100%.

**Comment by micate** [comment]
> Ok. Somehow I got them working again (although I don't have the most reliable procedure for reproducing).
> 
> I turned off two flags in the devtool experiments section:
> 
> * `Sync DevTools settings with Chrome Sync`  and
> * `Enable Reporting API panel in the Application panel`.
> 
> I also thought that doing a refresh of devtools could have worked, but it didn't appear to be consistent. I think turning off the first flag is probably the only thing, but I'm not 100%.

+1, and turned off these two flags have not worked for me.

**Comment by Hannah-goodridge** [comment]
> saga continues: https://bugs.chromium.org/p/chromium/issues/detail?id=1306249
> 
> ...which feels silly, because all of my extensions seem to load on _that_ page.

For those following this ticket, it has has been marked as duplicate and merged into this one:
https://bugs.chromium.org/p/chromium/issues/detail?id=1305475#c1

**Comment by istateside** [comment]
There still doesn't seem to be any movement on that Chrome bug, and I'm not able to provide a reproducible example to push it forward any more, but if folks are looking for a workaround:

- Open a new tab
- Open the Chrome devtools for your new tab
- navigate to whatever your React site's URL is in the new tab
- tada, the Components / Profile tabs should show up, and will stay there if you refresh the page.

If you try loading the devtools after you've already navigated to the React site, the tabs won't show up. I'm not familiar enough with writing extensions to know how this all works, but I was able to determine that, in the failing case, the listener in [background.js](https://github.com/facebook/react/blob/main/packages/react-devtools-extensions/src/background.js#L9) for `chrome.runtime.onConnect` never gets hit, and therefore the content script never gets installed.

I'm not sure what triggers that connection, but it doesn't happen unless you load the devtools before navigating to the React site.

It sounds like we've determined that it's an issue with Chrome, and not React, but hopefully this comment helps unblock someone else until the Chrome bug can be fixed.

**Comment by anushasingh02** [comment]
Very late to this but was still getting the bug. Had to toggle a random flag and refresh devtools which seems to be a fix.

**Comment by supriya-ck12** [comment]
> Very late to this but was still getting the bug. Had to toggle a random flag and refresh devtools which seems to be a fix.

Can you share which flag it was that had to be toggled?

**Comment by wonhotoss** [comment]
In my case, devtools - application - service workers - update on refresh was the reason. When it checked, react panels are not shown.

**Comment by alim-ansari** [comment]
> In my case, devtools - application - service workers - update on refresh was the reason. When it checked, react panels are not shown.

Yeah, same in my case. Fixed it.

**Comment by venku** [comment]
> In my case, devtools - application - service workers - update on refresh was the reason. When it checked, react panels are not shown.

Same with me. Fixed after unchecking

**Comment by AshfaqueSami2** [comment]
import react from '@vitejs/plugin-react'; // Import the Vite React plugin
import { defineConfig } from 'vite';

export default defineConfig({
  plugins: [react()], // Use the Vite React plugin
  build: {
    sourcemap: true, // Enable source maps generation
  },
});

@@***Use this in your vite.config.js everything wil work

**Comment by devVibe11** [comment]
For anyone who's having issues with the console tabs not appearing: like (Components)

Press F12 to open your console, click the gear/cog icon to open the console settings, then in the bottom right there's a button that says "Restore defaults and reload". This should bring the console tabs back if they're missing.

**Comment by devodii** [comment]
Still doesn't work for me.
I have an M2, 2022 model.
